### PR TITLE
adds new field `availability` to Course model

### LINKED ISSUE #1898: Course search: add "availability field" to course model
As a user searching for courses, I'd like to be able to see at a glance whether courses are available now, in the future, or in an archive. I'd also like to be able to filter my search results by availability. 

Currently, we calculate whether a course is available when building the index. but we found that we're missing some necessary dates for this calculation. 

Instead of using a calculation, we can rely on the `availability` field in the edX course API. All OCW course can be "Current" 

#### Acceptance Criteria:

- [ ] For MITx courses, import `availability` values from the course feed
- [ ] For OCW courses, all courses are "Current" 
- [ ] Add availability to the ES index

### PR DESCRIPTION
**Pull request by kyfantaz**
#### What are the relevant tickets?
Closes #1898 in conjunction with PR #1899. 
Note: This PR can be reviewed and approved independently, but PR #1899 should be merged first.

#### What's this PR do?
Adds new `availability` field to `course_catalog/Course` model.

#### How should this be manually tested?
You could run the newly added migration and check if the `Course` table has the appropriate column added in Postgres via manual inspection, and that the values match the `raw_json`.
```python
from course_catalog.models import *
for c in Course.objects.all(): print(c.platform, c.availability)
```

### CODE REVIEW

**Comment by mbertrand** [comment]
@kyfantaz running `docker-compose run web black course_catalog` should fix the formatting issue.

File: course_catalog/task_helpers.py [hunk 1]
```diff
@@ -431,3 +434,28 @@ def get_course_url(course_id, course_json, platform):
...
+    """
+    Gets the attribute `availability` for a course if any
+
+    Args:
+        course: Course model instance
+
+    Returns:
+        str: The url for the course if any
+    """
+    if course.platform == PlatformType.ocw.value:
```
Comment (mbertrand) [inline course_catalog/task_helpers.py]: Can you add unit tests to cover this function? ocw vs edx, edx with and without course runs

Comment (kyfantaz) [reply course_catalog/task_helpers.py]: :+1:

**Comment by mbertrand** [comment]
PS Code should be [squashed before merging into master](https://github.com/mitodl/handbook/blob/master/development-process.md#step-3-merging)

**Comment by kyfantaz** [comment]
:+1: thanks, I remember Alice's reminder on the last PR!

**Pull request merged**